internal/app/feed/exporter_report.go

```diff
@@ -0,0 +1,342 @@
+		DoUpdates: clause.AssignmentColumns([]string{"modified_at", "pdf", "word"}),
+	}).Create(&report)
+
+	if result.Error != nil {
+		e.Logger.Errorf("Failed to update save report status to local db for %s", inspection.Name)
+	}
+
+	return report
+}
+
+func (e *ReportExporter) exportInspection(ctx context.Context, apiClient api.APIClient, inspection *Inspection, format string) error {
+	mId, err := apiClient.InitiateInspectionReportExport(ctx, inspection.ID, format, e.PreferenceID)
+
+	if err != nil {
+		return err
+	}
+
+	tries := 0
+
+	for {
+		// wait for a second before checking for report completion
+		time.Sleep(1 * time.Second)
+		rec, cErr := apiClient.CheckInspectionReportExportCompletion(ctx, inspection.ID, mId)
+		if cErr != nil {
+			err = cErr
+			break
+		} else if rec.Status == "SUCCESS" {
+			resp, dErr := apiClient.DownloadInspectionReportFile(ctx, rec.URL)
+			if dErr != nil {
+				err = dErr
+				break
+			}
+
+			// only allow one process to access disk at the same time
+			// this way we won't allow process to overwrite reports with the same name
+			e.Mu.Lock()
+			err = saveReportResponse(resp, inspection, e.ExportPath, format)
+			e.Mu.Unlock()
+			break
+		} else if rec.Status == "FAILED" {
+			err = fmt.Errorf("%s report generation failed on server for %s", format, fmt.Sprintf("%s (%s)", inspection.Name, inspection.ID))
+			break
+		}
+
+		// make sure we stop checking after a while
+		tries += 1
+		if tries == 15 {
+			err = fmt.Errorf("%s report generation for %s terminated after %d tries", format, inspection.Name, tries)
+			break
+		}
+	}
+
+	return err
+}
+
+func (e *ReportExporter) updateReportResult(rep *Report, res *SaveReportsResult, inspection *Inspection, remaining int64) {
+	fn := fmt.Sprintf("%s (%s)", inspection.Name, inspection.ID)
+	if rep == nil {
+		res.NoChange += 1
+		e.Logger.Infof("No changes were made to %s", fn)
+	} else {
+		if rep.PDF == 1 {
+			res.PDFReports += 1
+			e.Logger.Infof("Saved PDF report for %s", fn)
+		} else if rep.PDF == -1 {
+			res.PDFErrors += 1
+		}
+
+		if rep.WORD == 1 {
+			res.WORDReports += 1
+			e.Logger.Infof("Saved Word report for %s", fn)
+		} else if rep.WORD == -1 {
+			res.WORDErrors += 1
+		}
+
+		e.Logger.Infof("%d inspections remaining", remaining)
+	}
+}
+
+func saveReportResponse(resp io.ReadCloser, inspection *Inspection, path string, format string) error {
+	filePath := getFilePath(path, inspection, format)
+	out, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+
+	_, err = io.Copy(out, resp)
+	resp.Close()
+	out.Close()
+	return err
+}
+
+func sanitizeName(name string) string {
+	res := strings.ReplaceAll(name, " / ", "-")
+	res = strings.ReplaceAll(res, " // ", "-")
+	res = strings.ReplaceAll(res, "/", "-")
+	res = strings.ReplaceAll(res, " ", "-")
+	return res
+}
+
+func getFileExtension(format string) string {
+	switch format {
+	case "PDF":
+		return "pdf"
+	case "WORD":
+		return "docx"
+	}
+	return ""
+}
```

Comment:
```suggestion
	case default:
	        return ""
        }
```

Reply:
applied the suggestion in my latest commit